javalib/src/main/scala/java/util/AbstractCollection.scala

```diff
@@ -0,0 +1,89 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.immutable.{ List => SList }
+
+abstract class AbstractCollection[E]() extends java.util.Collection[E] {
+  def iterator(): Iterator[E]
+  def size(): Int
+
+  @tailrec
+  private def iterateAndDo[X](i: Iterator[E], f: (E) => X, part: SList[X] = SList()): SList[X] =
```

Comment:
It seems this is essentially a `map`. Name it as such.

Actually, it seems that this method, and the others below, are basically methods of the Scala collections library. Why don't you simply import `scala.collection.JavaConversions.asScalaIterator` and be done with it?